src/Slist.hs

```diff
@@ -1872,3 +1880,90 @@ insert = insertBy compare
 insertBy :: (a -> a -> Ordering) -> a -> Slist a -> Slist a
 insertBy f a Slist{..} = Slist (L.insertBy f a sList) (sSize + 1)
 {-# INLINE insertBy #-}
+
+
+----------------------------------------------------------------------------
+-- Generic Functions
+----------------------------------------------------------------------------
+
+{- | @O(1)
+The 'genericLength' function is an overloaded version of 'length'. Instead
+of returning an 'Int', it returns any type which is an instance of 'Num'. 
+In case of an infinite list, 'genericLength' will return -1.
+-}
+genericLength :: Num i => Slist a -> i
+genericLength Slist{..} = case sSize of
+    Infinity -> -1
```

Comment:
This is not consistent with the library. I personally don't like this and would like to avoid such differences. I see that there is a way to avoid this.